core/node/state_keeper/src/mempool_actor.rs

```diff
@@ -131,6 +348,14 @@ impl MempoolFetcher {
                 .await
                 .context("failed syncing mempool")?;
 
+            let (transactions_with_constraints, transactions_to_return) =
+                filter_unsafe_deposits(transactions_with_constraints, &mut storage).await?;
+            storage
+                .transactions_dal()
+                .return_to_mempool(&transactions_to_return)
```

Comment:
Does transactionality with a `sync_mempool` above matter here? I.e., would anything bad happen if a node crashes after a `sync_mempool()`, but before this call?

Reply:
Agree, good point.